src/pages/TransactionPage/index.tsx

```diff
@@ -47,14 +48,25 @@ import {
   ContractTextarea,
   UTXOHashOutputSkip,
 } from './components';
+import { calculateTransactionFee } from './utils/gas';
 
 export default function TransactionPage() {
   const { transaction } = useParams() as any;
   const { data } = useTransactionPageQuery({ variables: { id: transaction } });
   const tx = data?.transaction;
+  const { chains } = useContext(ChainContext);
 
   if (!tx) return null;
 
+  const transactionFee = calculateTransactionFee({
+    bytePrice: +tx.bytePrice,
+    rawPayload: tx.rawPayload,
+    witnesses: tx.witnesses,
+    gasPriceFactor: +(chains[0].consensusParameters?.gasPriceFactor || 0),
+    gasPrice: +tx.gasPrice,
+    gasLimit: +tx.gasLimit,
```

Comment:
So, this is computing the _maximum_ fee the transaction will pay, not how much it actually paid. To compute that, you need to know the gas used (which was part of the original intention of this issue).

To get gas used, you need to read the last receipt of the transaction, and read the `gas_used` field: https://github.com/FuelLabs/fuel-specs/blob/204136996a795811907b472c88966dc09867e2a5/specs/protocol/abi.md#scriptresult-receipt